runtime/amplitude/src/lib.rs

```diff
@@ -146,13 +147,67 @@ pub type SignedPayload = generic::SignedPayload<RuntimeCall, SignedExtra>;
 /// Extrinsic type that has already been checked.
 pub type CheckedExtrinsic = generic::CheckedExtrinsic<AccountId, RuntimeCall, SignedExtra>;
 
+use crate::sp_api_hidden_includes_construct_runtime::hidden_include::dispatch::GetStorageVersion;
+use frame_support::pallet_prelude::StorageVersion;
+
+parameter_types! {
+	pub const InactiveAccounts: Vec<AccountId> = Vec::new();
+}
+
+// Temporary struct that defines the executions to be done upon upgrade,
+// Should be removed or at least checked on each upgrade to see if it is relevant,
+// given that these are "one-time" executions for particular upgrades
+pub struct CustomOnRuntimeUpgrade;
+impl frame_support::traits::OnRuntimeUpgrade for CustomOnRuntimeUpgrade {
+	fn on_runtime_upgrade() -> frame_support::weights::Weight {
+		log::info!("Custom on-runtime-upgrade function");
+
+		// WARNING: manually setting the storage version
+		if Contracts::on_chain_storage_version() == 0 {
+			log::info!("Upgrading pallet contract's storage version to 10");
+			StorageVersion::new(10).put::<Contracts>();
+		}
+		if Democracy::on_chain_storage_version() == 0 {
+			log::info!("Upgrading pallet democracy's storage version to 1");
+			StorageVersion::new(1).put::<Democracy>();
+		}
+		if Scheduler::on_chain_storage_version() == 3 {
+			log::info!("Upgrading pallet scheduler's storage version to 4");
+			StorageVersion::new(4).put::<Scheduler>();
+		}
+		if Preimage::on_chain_storage_version() == 0 {
+			log::info!("Upgrading pallet preimage's storage version to 1");
+			StorageVersion::new(1).put::<Preimage>();
+		}
+		if Multisig::on_chain_storage_version() == 0 {
+			log::info!("Upgrading pallet multisig's storage version to 1");
+			StorageVersion::new(1).put::<Multisig>();
+		}
+		if PolkadotXcm::on_chain_storage_version() == 0 {
+			log::info!("Upgrading pallet xcm's storage version to 1");
+			StorageVersion::new(1).put::<PolkadotXcm>();
+		}
+		if AssetRegistry::on_chain_storage_version() == 0 {
+			log::info!("Upgrading pallet asset registry's storage version to 2");
+			StorageVersion::new(2).put::<AssetRegistry>();
+		}
+
+		<Runtime as frame_system::Config>::DbWeight::get().reads_writes(8, 8)
```

Comment:
Nit: since we have if statements guarding the `put` of the storage versions, we will not always have `8` writes. 
We could add an extra variable for counting the amount of storage writes by incrementing it in each if statement. 

This is not super critical probably but once this runtime upgrade caused all migrations to execute and update the storage versions we'll always only have `8` reads and `0` writes making the returned weight inaccurate.

Reply:
True, although this code should not live more than this upgrade, it is better practice for sure.